server/rest/src/main/java/org/infinispan/rest/InvocationHelper.java

```diff
@@ -93,7 +92,7 @@ public ServerManagement getServer() {
 
    public EmbeddedCounterManager getCounterManager() {
       checkServerStatus();
-      return counterManager;
+      return (EmbeddedCounterManager) EmbeddedCounterManagerFactory.asCounterManager(protocolServer.getCacheManager());
```

Comment:
Should this check for null and throw IllegalLifecycleStateException if it is? Or this method is only ever guaranteed to be called after fully started?

Reply:
Internally, it will throw IllegalLifecycleStateException:

https://github.com/infinispan/infinispan/blob/9e1880ad090a6f7ca4508ca5a9264435cc01cd79/counter/src/main/java/org/infinispan/counter/EmbeddedCounterManagerFactory.java#L31-L32